src/payment/payment-method.ts

```diff
@@ -12,5 +11,5 @@ export default interface PaymentMethod {
     nonce?: string;
     initializationData?: any;
     returnUrl?: string;
-    initializationStrategy?: InitialisationStrategies;
+    initializationStrategy?: unknown;
```

Comment:
As the API grows over time `initializationStrategy` will effectively be a union of different interfaces.

At this point we do not know exactly how many or in what variety, so we've chosen to type as `unknown` and to defensively type-guard in later code to filter supported `initializationStrategy`(s).

Reply:
Unfortunately, `initializationStrategy` won't be a string, but a union of various objects. Something like:

```
interface BaseStrategy {
    type: string;
}

interface StrategyA extends BaseStrategy {
    // attributes and values specific to only StrategyA
}

interface StrategyB extends BaseStrategy {
    // attributes and values specific to only StrategyB
}

interface StrategyC extends BaseStrategy {
    // attributes and values specific to only StrategyC
}

type initializationStrategy = StrategyA | StrategyB | StrategyC;
```

And we don't know what the final union or shapes will be yet.

---

Typing it as union of what we know now, e.g.:

`type initializationStrategy = StrategyA | StrategyB`

Would lead to buggy behaviour once the same code also started receiving `StrategyC` from the updated (but not versioned) API.

Reply:
I've gone for 

```
interface UnknownObject {
    [key: string]: unknown;
}
export default interface InitializationStrategy extends Partial<UnknownObject> {
    type: string;
}
```

Which gives us the benefit of having `type` ensured, but also represents the 'yet to be clarified' nature of any other attributes. The latter paints a truer picture of the API and enforces later Typeguard use. (Without a union or `unknown` or `any` Typeguards would be out of place - only explained with domain knowledge).

It also means typeguards can interrogate possible properties on a strategy without type errors or needing to be overly reliant on just the `type` attribute. The one draw back was a slight 'hack' of using `pick` on the None typeguard (https://github.com/bigcommerce/checkout-sdk-js/pull/1150/files#diff-4dbdf09949352589db42a49b9d883d5c1f3713df9d0c4de312ba575dd8cd5cf8R6), but this should be the only exception.

What do you recon? I'm on the pessimistic/defensive side of Typing 😄